lib/event-bus/src/rabbit-event-bus/index.test.ts

```diff
@@ -0,0 +1,325 @@
+      const subscribeMock = jest.fn();
+      const connectMock = jest.fn();
+
+      // tslint:disable-next-line: no-any
+      (AMQPConnector as any).mockImplementation(
+        (_0, [send, _1]: Channel<StateChange<{}>>, _2, subscriptions) => {
+          send({
+            newState: 'NOT_CONNECTED',
+          });
+
+          return {
+            subscriptions,
+            connect: connectMock,
+            publish: jest.fn(() => false),
+            subscribe: subscribeMock,
+          };
+        },
+      );
+
+      const manager = await new RabbitEventBus({ url: '' }).init([], '');
+      const then = jest.fn();
+
+      manager
+        .publish({
+          kind: 'test',
+          namespace: 'test',
+          id: 'something',
+          created: new Date(),
+          payload: {},
+        })
+        .then(then);
+
+      manager
+        .publish({
+          kind: 'test',
+          namespace: 'test',
+          id: 'something',
+          created: new Date(),
+          payload: {},
+        })
+        .then(then);
+
+      manager
+        .publish({
+          kind: 'test',
+          namespace: 'test',
+          id: 'something',
+          created: new Date(),
+          payload: {},
+        })
+        .then(then);
+
+      setTimeout(() => {
+        expect(then).toHaveBeenCalledTimes(0);
+        done();
+      }, 50);
+    });
+
+    it('publish promises are resolved after a successful connection', async done => {
+      const subscribeMock = jest.fn();
+      const connectMock = jest.fn();
+
+      // tslint:disable-next-line: no-any
+      (AMQPConnector as any).mockImplementation(
+        (_0, [send, _1]: Channel<StateChange<{}>>, _2, subscriptions) => {
+          send({
+            newState: 'CONNECTED',
+          });
+
+          return {
+            subscriptions,
+            publish: jest.fn(() => true),
+            subscribe: subscribeMock,
+          };
+        },
+      );
+
+      const manager = await new RabbitEventBus({ url: '' }).init([], '');
+      const then = jest.fn();
+
+      manager
+        .publish({
+          kind: 'test',
+          namespace: 'test',
+          id: 'something',
+          created: new Date(),
+          payload: {},
+        })
+        .then(then);
+
+      manager
+        .publish({
+          kind: 'test',
+          namespace: 'test',
+          id: 'something',
+          created: new Date(),
+          payload: {},
+        })
+        .then(then);
+
+      manager
+        .publish({
+          kind: 'test',
+          namespace: 'test',
+          id: 'something',
+          created: new Date(),
+          payload: {},
+        })
+        .then(then);
+
+      setTimeout(() => {
+        expect(then).toHaveBeenCalledTimes(3);
+        done();
+      }, 250);
+    });
+  });
+
+  // Tests that it queues up messages that it can't send and that resends then when it can
+  // If a message fails to send it will retry it
```

Comment:
Shall we add these tests now or in a followup?

Reply:
Ah - if you get an approval from me with comments then I'm happy it go in now or later - happy its up to you.